pkg/placement/streams.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package placement
+
+import (
+	"strings"
+	"sync"
+	"sync/atomic"
+
+	"google.golang.org/grpc/metadata"
+
+	placementv1pb "github.com/dapr/dapr/pkg/proto/placement/v1"
+)
+
+type daprdStream struct {
+	id            uint32
+	hostName      string
+	hostID        string
+	hostNamespace string
+	needsVNodes   bool
+	stream        placementv1pb.Placement_ReportDaprStatusServer
+}
+
+func newDaprdStream(host *placementv1pb.Host, stream placementv1pb.Placement_ReportDaprStatusServer) *daprdStream {
+	return &daprdStream{
+		hostID:        host.GetId(),
+		hostName:      host.GetName(),
+		hostNamespace: host.GetNamespace(),
+		stream:        stream,
+		needsVNodes:   hostNeedsVNodes(stream),
+	}
+}
+
+// streamConnPool has the stream connections established between the placement gRPC server
+// and the Dapr runtime grouped by namespace, with an assigned id for faster lookup/deletion
+// The id is a simple auto-incrementing number, for efficiency.
+type streamConnPool struct {
```

Comment:
nit, should the conn pool logic be pulled out to a `pool.go` file?

Reply:
Yeah, the whole file is exclusive to stream conn pool logic.

Reply:
Can we still be changed to `pool.go` for consistency sake?  So far we have `grpc/manager/pool.go` and scheduler has a `pool.go`. It would be awesome if we could be consistent with our file naming conventions :)

Reply:
Of course, that makes sense. I'll change it 👍